dashboard/app/assets/javascripts/client_state.js

```diff
@@ -180,4 +180,35 @@ function hasSeenVisualElement(visualElementType, visualElementId) {
   }
 }
 
+/**
+ * Queries the server for updated script progress for the current user and
+ * merges it into the current clientState.
+ *
+ * @param scriptId
+ * @returns jquery request object
+ */
+dashboard.clientState.refreshScriptProgressFromServer = function (scriptId) {
+  var req = $.post("/milestone/script_progress/" + scriptId);
+  req.done(function (responseJSON) {
```

Comment:
I'm not so familiar with the new promise-style Ajax requests in jQuery. 

I was worried about the possibility that the post request might complete before the .done callback is attached, leading to a missed callback.  Hopefully the threading model in JavaScript is precisely defined and observed enough that this is guaranteed not to happen in all browsers?